### PR TITLE
Conversion between a non-nullable and nullable type can throw an exce…

### PR DESCRIPTION
**Pull request by JvanderStad**
I had an exception while converting an `Int32` to a `Nullable<Int64>`. The `TrySetValue` method now converts the prop/field value with method `ChangeType`.

`ChangeType` checks if the target type is a nullable and tries to change the type to the underlaying type of the nullable type

### CODE REVIEW

File: src/Scriban/Runtime/Accessors/TypedObjectAccessor.cs [hunk 1]
```diff
@@ -95,20 +95,32 @@ public bool TrySetItem(TemplateContext context, SourceSpan span, object target,
...
-            MemberInfo memberAccessor;
-            if (_members.TryGetValue(member, out memberAccessor))
+            if (_members.TryGetValue(member, out MemberInfo memberAccessor))
             {
-                var fieldAccessor = memberAccessor as FieldInfo;
-                if (fieldAccessor != null)
+                if (memberAccessor is FieldInfo fieldAccessor)
                 {
-                    fieldAccessor.SetValue(target, value);
+                    fieldAccessor.SetValue(target, ChangeType(value, fieldAccessor.FieldType));
```
Comment (xoofx) [inline src/Scriban/Runtime/Accessors/TypedObjectAccessor.cs:118]: Conversion between value to a new type should use `context.ToObject` instead. `ChangeType` should not be introduced

Comment (JvanderStad) [reply src/Scriban/Runtime/Accessors/TypedObjectAccessor.cs]: Righto. Updated

**Comment by JvanderStad** [comment]
@xoofx Any chance you could merge this PR? Thanks! 👍

**Comment by xoofx** [comment]
Yep, sorry, thanks!

**Pull request merged**